### ISSUE TITLE
[FEA] multiGPU support for metrics

### ISSUE THREAD

**Issue by bschifferer** [NVIDIA-Merlin/NVTabular#1364]
**Is your feature request related to a problem? Please describe.**
We use TensorFlow+Keras+Horovod for multi-GPU examples. Some evaluation metrices cannot be averages, for example AUC. As a user, I want to calculate a correct AUC over multiple GPUs.

**Additional context**
 I found this for TF1.x https://github.com/horovod/horovod/issues/444
Or https://stackoverflow.com/questions/56552397/custom-metric-using-scikit-learns-aucroc-calculator-with-tf-keras

The 2nd one suggest to implement a custom callback.
